pkg/services/validator/aws.go

```diff
@@ -57,44 +68,421 @@ func readAwsPlugin(vc *components.ValidatorConfig, k8sClient kubernetes.Interfac
 		return err
 	}
 
-	if !c.IamCheck.Enabled && c.Validator.ResultCount() == 0 {
+	if c.Validator.ResultCount() == 0 {
 		return errNoRulesEnabled
 	}
 	return nil
 }
 
-func configureIamRule(c *components.AWSPluginConfig) error {
+func configureIamRoleRules(c *components.AWSPluginConfig, ruleNames *[]string) error {
 	log.InfoCLI(`
-	AWS IAM validation ensures that a certain IAM Role has every permission
-	specified in one of Spectro Cloud's predefined IAM policies.
+	AWS IAM Role validation ensures that specified IAM Roles have every permission
+	specified in the passed in policy documents.
+	`)
+
+	validateRoles, err := prompts.ReadBool("Enable Iam Role validation", true)
+	if err != nil {
+		return err
+	}
+	if !validateRoles {
+		c.Validator.IamRoleRules = nil
+		return nil
+	}
+	for i, r := range c.Validator.IamRoleRules {
+		r := r
+		if err := readIamRoleRule(c, &r, i, ruleNames); err != nil {
+			return err
+		}
+	}
+	addRules := true
+	if c.Validator.IamRoleRules == nil {
+		c.Validator.IamRoleRules = make([]vpawsapi.IamRoleRule, 0)
+	} else {
+		addRules, err = prompts.ReadBool("Add another role rule", false)
+		if err != nil {
+			return err
+		}
+	}
+	if !addRules {
+		return nil
+	}
+	for {
+		if err := readIamRoleRule(c, nil, -1, ruleNames); err != nil {
+			return err
+		}
+		add, err := prompts.ReadBool("Add another role rule", false)
+		if err != nil {
+			return err
+		}
+		if !add {
+			break
+		}
+	}
+	return nil
+}
+
+func readIamRoleRule(c *components.AWSPluginConfig, r *vpawsapi.IamRoleRule, idx int, ruleNames *[]string) error {
+	if r == nil {
+		r = &vpawsapi.IamRoleRule{
+			IamRoleName: "",
+			Policies:    []vpawsapi.PolicyDocument{},
+		}
+	}
+	err := initAwsRule(r, "iam role", ruleNames)
+	if err != nil {
+		return err
+	}
+	if r.IamRoleName == "" {
+		roleName, err := prompts.ReadText("IAM Role Name", "", false, -1)
+		if err != nil {
+			return err
+		}
+		r.IamRoleName = roleName
+	}
+
+	addPolicies := true
+	for addPolicies {
+		policyDoc, err := readIamPolicy()
+		if err != nil {
+			return err
+		}
+
+		r.Policies = append(r.Policies, policyDoc)
+		addPolicies, err = prompts.ReadBool("Add another policy document", false)
+		if err != nil {
+			return err
+		}
+	}
+	if idx == -1 {
+		c.Validator.IamRoleRules = append(c.Validator.IamRoleRules, *r)
+	} else {
+		c.Validator.IamRoleRules[idx] = *r
+	}
+	return nil
+}
 
-	Different permission sets are required to deploy clusters via Spectro Cloud,
-	depending on placement type (static vs. dynamic) and other factors.
+func configureIamUserRules(c *components.AWSPluginConfig, ruleNames *[]string) error {
+	log.InfoCLI(`
+	AWS IAM User validation ensures that specified IAM Users have every permission
+	specified in the passed in policy documents.
```

Comment:
nit: `in the provided policy document(s).` here and in the 3 other spots if you don't mind 🙇🏼